scripts/execute_cypress.sh

```diff
@@ -0,0 +1,161 @@
+#!/bin/bash
+
+# Exit immediately if a command exits with a non-zero status,
+# Treat unset variables as an error, and prevent errors in a pipeline from being masked
+set -euo pipefail
+
+# define colors
+RESET='\033[0m'
+RED='\033[0;31m'
+YELLOW='\033[0;33m]'
+
+# Define arrays for services, etc.
+# Read service arrays from environment variables
+read -r -a payments <<< "${PAYMENTS[@]}"
+read -r -a payouts <<< "${PAYOUTS[@]}"
+read -r -a payment_method_list <<< "${PAYMENT_METHOD_LIST[@]}"
+read -r -a routing <<< "${ROUTING[@]}"
+
+# define arrays
+connector_map=()
+failed_connectors=()
+
+# Define an associative array to map environment variables to service names
+declare -A services=(
+  ["PAYMENTS"]="payments"
+  ["PAYOUTS"]="payouts"
+  ["PAYMENT_METHOD_LIST"]="payment_method_list"
+  ["ROUTING"]="routing"
+)
+
+# Function to read service arrays from environment variables
+function read_service_arrays() {
+  # Loop through the associative array and check if each service is exported
+  for var in "${!services[@]}"; do
+    if [[ -n "${!var+x}" ]]; then
+      connector_map+=("${services[$var]}")
+    fi
+  done
+}
+
+# Function to print messages in color
+function print_color() {
+  local color="$1"
+  local message="$2"
+  echo -e "${color}${message}${RESET}"
+}
+
+# Function to check if a command exists
+function command_exists() {
+  command -v "$1" > /dev/null 2>&1
+}
+
+# Function to execute Cypress tests
+function execute_test() {
+  local connector="$1"
+  local service="$2"
+  local tmp_file="$3"
+
+  print_color "YELLOW" "Executing tests for ${service} with connector ${connector}..."
+
+  export REPORT_NAME="${service}_${connector}_report"
+  if ! CYPRESS_CONNECTOR="$connector" npm run "cypress:$service"; then
+    echo "${service}-${connector}" >> "$tmp_file"
```

Comment:
we write failed connectors to a tmp file because `execute_test` function runs within a subshell and data is not accessible outside of it, so we write it to a file instead.